model/task/task.go

```diff
@@ -909,3 +915,68 @@ func ExpectedTaskDuration(project, buildvariant string, window time.Duration) (m
 
 	return expDurations, nil
 }
+
+// MergeNewTestResults returns the task with both old (embedded in
+// the tasks collection) and new (from the testresults collection) test results
+// merged in the Task's TestResults field.
+func (t *Task) MergeNewTestResults() error {
+	collection := Collection
+	lookupLocalField := IdKey
+	if t.Archived {
+		collection = OldCollection
+		lookupLocalField = OldTaskIdKey
+	}
+
+	// aggregation stages
+	matchStage := bson.M{"$match": bson.M{IdKey: t.Id}}
+	lookupStage := bson.M{"$lookup": bson.M{
+		"from":         testresult.Collection,
+		"localField":   lookupLocalField,
+		"foreignField": testresult.TaskIDKey,
+		"as":           newresultsAllExecutionsField,
+	}}
+	addNewResultsStage := bson.M{"$addFields": bson.M{
+		newresultsCurrentExecutionsField: bson.M{
+			"$filter": bson.M{
+				"input": "$" + newresultsAllExecutionsField,
+				"as":    "testresult",
+				"cond": bson.M{
+					"$eq": []string{
+						"$$" + "testresult" + "." + testresult.ExecutionKey,
+						"$" + ExecutionKey,
+					},
+				},
+			},
+		},
+	}}
+	concatStage := bson.M{"$addFields": bson.M{
```

Comment:
If you're concating the arrays are you worried about duplicate entries? The only scenario I can think of when we'd get in to this situation is if we have both the old and new schemas at the same time but idk if we need to handle this

Reply:
No, we shouldn't ever have duplicates.